<?xml version="1.0" ?>
<launch>
  <arg name="node_name"             default="velocity_smoother"/>
  <arg name="nodelet_manager_name"  default="nodelet_manager"/>
  <arg name="raw_cmd_vel_topic"     default="raw_cmd_vel"/>
  <arg name="smooth_cmd_vel_topic"  default="smooth_cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   default="robot_cmd_vel"/>
  <arg name="odom_topic"            default="odom"/>

  <arg name="SPEED_LIM_V"            default="0.8"/>
  <arg name="SPEED_LIM_W"            default="5.4"/>
  <arg name="ACCEL_LIM_V"            default="0.3"/>
  <arg name="ACCEL_LIM_W"            default="3.5"/>
  <arg name="FREQUENCY"            default="20.0"/>
  <arg name="DECEL_FACTOR"            default="1.0"/>
  <arg name="ROBOT_FEEDBACK"            default="2"/>



  
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"/>

  <node pkg="nodelet" type="nodelet" name="$(arg node_name)"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet $(arg nodelet_manager_name)">
        
      <param name="speed_lim_v" type="double" value="$(arg SPEED_LIM_V)" />
      <param name="speed_lim_w" type="double" value="$(arg SPEED_LIM_W)" />
      <param name="accel_lim_v" type="double" value="$(arg ACCEL_LIM_V)" />
      <param name="accel_lim_w" type="double" value="$(arg ACCEL_LIM_W)" />
      <param name="frequency" type="double" value="$(arg FREQUENCY)" />
      <param name="decel_factor" type="double" value="$(arg DECEL_FACTOR)" />
      <param name="robot_feedback" type="int" value="$(arg ROBOT_FEEDBACK)" />
    <!-- velocity commands I/O -->
    <remap from="$(arg node_name)/raw_cmd_vel"    to="$(arg raw_cmd_vel_topic)"/>
    <remap from="$(arg node_name)/smooth_cmd_vel" to="$(arg smooth_cmd_vel_topic)"/>

    <!-- Robot velocity feedbacks -->
    <remap from="$(arg node_name)/robot_cmd_vel"  to="$(arg robot_cmd_vel_topic)"/>
    <remap from="$(arg node_name)/odometry"       to="$(arg odom_topic)"/>
  </node>
</launch>
